optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/common/DroolsUniAccumulateFunctionAdapter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.drools.common;
+
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.kie.api.runtime.rule.AccumulateFunction;
+import org.optaplanner.core.api.score.stream.uni.UniConstraintCollector;
+import org.optaplanner.core.impl.score.stream.drools.uni.UniConstraintCollectorAdapter;
+
+/**
+ * Drools {@link AccumulateFunction} that calls {@link UniConstraintCollector} underneath.
+ * @param <A> input to accumulate
+ * @param <ResultContainer_> implementation detail
+ * @param <NewA> result of accumulation
+ */
+public final class DroolsUniAccumulateFunctionAdapter<A, ResultContainer_ extends Serializable, NewA>
+        implements AccumulateFunction<ResultContainer_> {
```

Comment:
Design patterns: I 'd argue it's a Bridge, not an Adapter pattern.
https://stackoverflow.com/questions/1425171/difference-between-bridge-pattern-and-adapter-pattern
But that's like the difference between an ale and a lager: most people will happily interchange them.